### PR TITLE
update lm_eval version

### PR DESCRIPTION
**Pull request by alexey-belyakov**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Comment by jiminha** [comment]
Can you clarify which tests have been run and validated with this upgrade? Also please update the Description details on this PR.

**Review by FocusLuo — APPROVED** [review]
LGTM,
verified this patches on doing rebase on the latest commit c6c2d6db4590d24feb6d03089ee1476301c344e1
On, the Gaudi2
python run_lm_eval.py --model_name_or_path /data/huggingface/hub/models--meta-llama--Meta-Llama-3-8B/snapshots/8cde5ca8380496c9a6cc7ef3a8b46a0372a1d920/ --use_hpu_graphs --use_kv_cache --bf16 --batch_size=8 --tasks leaderboard_mmlu_pro -o eval.json

the result is expected:
 "results": {
    "leaderboard_mmlu_pro": {
      "alias": "leaderboard_mmlu_pro",
      "acc,none": 0.3081781914893617,
      "acc_stderr,none": 0.004209661852408173
    }
  },
  "group_subtasks": {
    "leaderboard_mmlu_pro": []
  },

**Comment by FocusLuo** [comment]
root@gaudi2-2:/workspace/optimum-habana/examples/text-generation# python run_lm_eval.py --model_name_or_path /data/huggingface/hub/models--meta-llama--Meta-Llama-3-8B/snapshots/8cde5ca8380496c9a6cc7ef3a8b46a0372a1d920/ --use_hpu_graphs --use_kv_cache --bf16 --batch_size=8 --tasks leaderboard_mmlu_pro -o eval-2.json
/usr/local/lib/python3.10/dist-packages/torch/distributed/distributed_c10d.py:366: UserWarning: torch.distributed.reduce_op is deprecated, please use torch.distributed.ReduceOp instead
  warnings.warn(
[WARNING|utils.py:212] 2024-11-26 14:54:39,344 >> optimum-habana v1.15.0.dev0 has been validated for SynapseAI v1.18.0 but habana-frameworks v1.17.0.495 was found, this could lead to undefined behavior!
/usr/local/lib/python3.10/dist-packages/transformers/deepspeed.py:24: FutureWarning: transformers.deepspeed module is deprecated and will be removed in a future version. Please import deepspeed modules directly from transformers.integrations
  warnings.warn(
/usr/local/lib/python3.10/dist-packages/habana_frameworks/torch/hpu/__init__.py:160: UserWarning: torch.hpu.setDeterministic is deprecated and will be removed in next release. Please use torch.use_deterministic_algorithms instead.
  warnings.warn(
[2024-11-26 14:54:41,330] [INFO] [real_accelerator.py:219:get_accelerator] Setting ds_accelerator to hpu (auto detect)
2024-11-26:14:54:42,441 INFO     [utils.py:85] DeepSpeed is enabled.
[2024-11-26 14:54:42,442] [INFO] [comm.py:652:init_distributed] cdb=None
[2024-11-26 14:54:42,442] [INFO] [comm.py:667:init_distributed] Not using the DeepSpeed or dist launchers, attempting to detect MPI environment...
[2024-11-26 14:54:42,937] [INFO] [comm.py:718:mpi_discovery] Discovered MPI settings of world_rank=0, local_rank=0, world_size=1, master_addr=169.254.3.1, master_port=29500
[2024-11-26 14:54:42,937] [INFO] [comm.py:683:init_distributed] Initializing TorchBackend in DeepSpeed with backend hccl
[2024-11-26 14:54:43,144] [INFO] [logging.py:128:log_dist] [Rank 0] DeepSpeed info: version=0.16.0, git-hash=unknown, git-branch=unknown
[2024-11-26 14:54:43,147] [INFO] [logging.py:128:log_dist] [Rank 0] quantize_bits = 8 mlp_extra_grouping = False, quantize_groups = 1
Loading 4 checkpoint shards:   0%|                                                              | 0/4 [00:00<?, ?it/s]============================= HABANA PT BRIDGE CONFIGURATION ===========================
 PT_HPU_LAZY_MODE = 1
 PT_RECIPE_CACHE_PATH =
 PT_CACHE_FOLDER_DELETE = 0
 PT_HPU_RECIPE_CACHE_CONFIG =
 PT_HPU_MAX_COMPOUND_OP_SIZE = 9223372036854775807
 PT_HPU_LAZY_ACC_PAR_MODE = 0
 PT_HPU_ENABLE_REFINE_DYNAMIC_SHAPES = 0
---------------------------: System Configuration :---------------------------
Num CPU Cores : 152
CPU RAM       : 1056438812 KB
------------------------------------------------------------------------------
Loading 4 checkpoint shards: 100%|██████████████████████████████████████████████████████| 4/4 [00:05<00:00,  1.39s/it]
2024-11-26:14:54:49,691 INFO     [utils.py:85] Args: Namespace(buckets=[16, 32, 64, 128, 189, 284], output_file='eval-2.json', tasks=['leaderboard_mmlu_pro'], limit_iters=None, device='hpu', model_name_or_path='/data/huggingface/hub/models--meta-llama--Meta-Llama-3-8B/snapshots/8cde5ca8380496c9a6cc7ef3a8b46a0372a1d920/', bf16=True, max_new_tokens=100, max_input_tokens=0, batch_size=8, warmup=3, n_iterations=5, local_rank=0, use_kv_cache=True, use_hpu_graphs=True, dataset_name=None, column_name=None, do_sample=False, num_beams=1, top_k=None, penalty_alpha=None, trim_logits=False, seed=27, profiling_warmup_steps=0, profiling_steps=0, profiling_record_shapes=False, prompt=None, bad_words=None, force_words=None, assistant_model=None, peft_model=None, num_return_sequences=1, token=None, model_revision='main', attn_softmax_bf16=False, output_dir=None, bucket_size=-1, bucket_internal=False, dataset_max_samples=-1, limit_hpu_graphs=False, show_graphs_count=False, reuse_cache=False, verbose_workers=False, simulate_dyn_prompt=None, reduce_recompile=False, use_flash_attention=False, flash_attention_recompute=False, flash_attention_causal_mask=False, flash_attention_fast_softmax=False, book_source=False, torch_compile=False, ignore_eos=True, temperature=1.0, top_p=1.0, const_serialization_path=None, trust_remote_code=False, parallel_strategy='none', input_embeds=False, run_partial_dataset=False, load_quantized_model_with_autogptq=False, disk_offload=False, load_quantized_model_with_inc=False, local_quantized_inc_model_path=None, quant_config='', world_size=1, global_rank=0)
2024-11-26:14:54:49,691 INFO     [utils.py:85] device: hpu, n_hpu: 1, bf16: True
2024-11-26:14:54:49,691 INFO     [utils.py:85] Model initialization took 8.912s
Starting from v4.46, the `logits` model output will have the same type as the model (except at train time, where it will always be FP32)
100%|████████████████████████████████████████████████████████████████████████| 12032/12032 [00:00<00:00, 14848.27it/s]
Running loglikelihood requests:   5%|█▉                                        | 5136/113995 [00:18<06:18, 287.90it/s]

**Comment by hsubramony** [comment]
could you please submit this patch to OHF as well so that we can make sure its reported in kibana.
thanks

**Comment by regisss** [comment]
@alexey-belyakov Please sync your branch with main and run `make style`

**Comment by regisss** [comment]
> Hi @regisss !
> 
> Thank you for updating the branch, unfortunately @alexey-belyakov is not available right now. I was creating a new PR with main syncing and we need to modify readme a little. I'll appreciate if you can add this change for us. If it's not ok, I can create a new PR.
> 
> I'll put the change in the review.
> 
> Thanks, Dmitry

Sure @dsmertin, I'll add it to this PR :slightly_smiling_face:

**Comment by regisss** [comment]
@dsmertin I added the change you suggested, let me know if that looks good to you

**Review by regisss — APPROVED** [review]
